apps/opik-backend/src/main/java/com/comet/opik/domain/llmproviders/LlmProviderFactory.java

```diff
@@ -0,0 +1,58 @@
+package com.comet.opik.domain.llmproviders;
+
+import com.comet.opik.api.LlmProvider;
+import com.comet.opik.domain.LlmProviderApiKeyService;
+import com.comet.opik.infrastructure.EncryptionUtils;
+import com.comet.opik.infrastructure.LlmProviderClientConfig;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.BadRequestException;
+import lombok.NonNull;
+import ru.vyarus.dropwizard.guice.module.yaml.bind.Config;
+
+@Singleton
+public class LlmProviderFactory {
+    private final LlmProviderClientConfig llmProviderClientConfig;
+    private final LlmProviderApiKeyService llmProviderApiKeyService;
+
+    @Inject
+    public LlmProviderFactory(
+            @NonNull @Config LlmProviderClientConfig llmProviderClientConfig,
+            @NonNull LlmProviderApiKeyService llmProviderApiKeyService) {
+        this.llmProviderApiKeyService = llmProviderApiKeyService;
+        this.llmProviderClientConfig = llmProviderClientConfig;
+    }
+
+    public LlmProviderService getService(@NonNull String workspaceId, @NonNull String model) {
+        var llmProvider = getLlmProvider(model);
+        if (llmProvider == LlmProvider.OPEN_AI) {
+            var apiKey = EncryptionUtils.decrypt(getEncryptedApiKey(workspaceId, llmProvider));
```

Comment:
this is not specific to open AI, should be outside the if / switch case.

Reply:
Done in https://github.com/comet-ml/opik/pull/957/commits/05c74ff22517cf9803a0c02aa73ba030ae542d35